common/ferrostar/src/routing_adapters/osrm/models.rs

```diff
@@ -500,4 +535,88 @@ mod tests {
             Some(vec!["right".to_string()])
         );
     }
+
+    #[test]
+    fn deserialize_incidents() {
+        let data = r#"
+        {
+          "id": "13956787949218641",
+          "type": "construction",
+          "creation_time": "2024-11-13T16:39:17Z",
+          "start_time": "2023-04-03T22:00:00Z",
+          "end_time": "2024-11-26T04:59:00Z",
+          "iso_3166_1_alpha2": "US",
+          "iso_3166_1_alpha3": "USA",
+          "description": "I-84 W/B: intermittent lane closures from Exit 57 CT-15 to US-44 Connecticut Blvd",
+          "long_description": "Intermittent lane closures due to barrier repairs on I-84 Westbound from Exit 57 CT-15 to US-44 Connecticut Blvd.",
+          "impact": "major",
+          "sub_type": "CONSTRUCTION",
+          "alertc_codes": [
+            701,
+            703
+          ],
+          "traffic_codes": {
+            "incident_primary_code": 701
+          },
+          "lanes_blocked": [],
+          "length": 2403,
+          "south": 41.763362,
+          "west": -72.661148,
+          "north": 41.769363,
+          "east": -72.633712,
+          "congestion": {
+            "value": 101
+          },
+          "geometry_index_start": 2932,
+          "geometry_index_end": 3017,
+          "affected_road_names": [
+            "Officer Brian A. Aselton Memorial Highway"
+          ],
+          "affected_road_names_unknown": [
+            "I 84 West/US 6"
+          ],
+          "affected_road_names_en": [
+            "Officer Brian A. Aselton Memorial Highway"
+          ]
+          }
+        "#;
+
+        let incident: OsrmIncident = serde_json::from_str(data).expect("Failed to parse Incident");
+
+        // help me finish this test
```

Comment:
I think we can simplify all this manual work with a snapshot test; I'll handle that real quick.